packages/core/src/init.ts

```diff
@@ -1,24 +1,61 @@
 import { getGPUTier } from 'detect-gpu';
 import { SharedArrayBufferModes } from './enums';
-
 let csRenderInitialized = false;
-let useCPURendering = false;
 let useSharedArrayBuffer = true;
-let sharedArrayBufferMode = SharedArrayBufferModes.TRUE;
+let sharedArrayBufferMode = SharedArrayBufferModes.AUTO;
+import { deepMerge } from './utilities';
+import { Cornerstone3DConfig } from './types';
+// TODO: move sharedArrayBuffer into config.
 
-// https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/By_example/Detect_WebGL
-function hasActiveWebGLContext() {
+const defaultConfig = {
+  detectGPU: {},
+  rendering: {
+    preferSizeOverAccuracy: true,
+    useCPURendering: false,
+    hasNorm16TextureSupport: _hasNorm16TextureSupport(),
+  },
+  // cache
+  // ...
+};
+
+let config = {
+  detectGPU: {},
+  rendering: {
+    preferSizeOverAccuracy: true,
+    useCPURendering: false,
+    hasNorm16TextureSupport: _hasNorm16TextureSupport(),
+  },
```

Comment:
@sedghi based on our discussion, this would default to false for now for both preferSizeOverAccuracy and hasNorm16TextureSupport

Reply:
I think we should have the hasNorm16TextureSupport settable, which make me think we should rename it to `useNorm16Texture` since it has bugs in browsers and user can set it as necessary